### PR TITLE
Refactor trigger verbose debug report generation in spec

### PR DESCRIPTION
**Pull request by linnan-github**
Moves the debug data generation to `triggering attribution` algorithm for simplification.

***
<a href="https://pr-preview.s3.amazonaws.com/linnan-github/conversion-measurement-api/pull/1285.html" title="Last updated on May 17, 2024, 6:14 PM UTC (4f5d359)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/1285/4565ada...linnan-github:4f5d359.html" title="Last updated on May 17, 2024, 6:14 PM UTC (4f5d359)">Diff</a>

### CODE REVIEW

File: index.bs
Comment (apasel422) [inline index.bs:3427]: What do you think about making this a list instead of a set? I ask because it's not clear to me how the equality operation is supposed to work for `trigger debug data`, particularly with respect to its `report` field.

File: index.bs
Comment (apasel422) [inline index.bs:3435]: Instead of passing the data type and report as separate arguments, should we just pass `data` itself?

File: index.bs
Comment (apasel422) [inline index.bs:3473]:
```suggestion
        |trigger|, and [=obtain and deliver a debug report on trigger registration/sourceToAttribute=] set to null.
```

**Review by apasel422 — CHANGES_REQUESTED** [review]

File: index.bs
Comment (linnan-github) [reply index.bs:3427]: The comparison is field-by-field, including the report field, which is mostly null except for two event errors. So basically it’s just comparing types.

File: index.bs
Comment (linnan-github) [reply index.bs:3435]: Passing the data type and report is more flexible if data generation is different.

File: index.bs
Comment (apasel422) [reply index.bs:3427]: Is there actually a benefit to using a set though? As written, it seems like implementations will need to compare every single field in the report in this algorithm.

Comment (linnan-github) [reply index.bs:3427]: As event-level and aggregate may have the same debug type, so using a set will deduplicate them and only results in one payload. Alternately, we can pass as a list and deduplicate based on types in the report generation algorithm. Wdyt?

Comment (apasel422) [reply index.bs:3427]: Could you clarify when they can have the same debug type? I thought that the common debug types were only generated before calling the aggregatable-specific and event-level-specific algorithms.

Comment (apasel422) [reply index.bs:3427]: I think it would be clearer if we refactored [that algorithm](https://pr-preview.s3.amazonaws.com/linnan-github/conversion-measurement-api/pull/1285.html#check-if-attribution-should-be-blocked-by-rate-limits) to perform [should processing be blocked by reporting-origin limit](https://pr-preview.s3.amazonaws.com/linnan-github/conversion-measurement-api/pull/1285.html#should-block-processing-for-reporting-origin-limit) only once per attribution, instead of once for event-level and once for aggregatable, since IIUC the result should be the same.

If that's not possible because the checks are done in a particular order for privacy reasons, then I would expect that algorithm to have a note making that clear.

Comment (linnan-github) [reply index.bs:3431]: Added an issue to considering refactoring to make this PR no behavioral change.

Comment (apasel422) [reply index.bs:3431]: Could you also add an issue for changing the type from `set` to `list`?

**Review by apasel422 — CHANGES_REQUESTED** [review]
LGTM once the issue is added.

File: index.bs
Comment (linnan-github) [reply index.bs:3431]: Added an issue, thanks.

**Review by apasel422 — APPROVED** [review]

**Pull request merged**
